### PR TITLE
#15905 add partitions statements to the table DDL for PostgreSQL

### LINKED ISSUE #15905: DDL for partitions for partitioned tables on postgresql 10+
#### System information: 
- Operating system: any

#### Connection specification:
- PostgreSQL 10+ (partition support)

#### Describe the problem you're observing:
DDL for partitioned table does not include partitions DDL. PostgreSQL does not create partitions automatically
image
Expected: DDL settings to have partitions checkbox to (not) include partitions DDL for partitioned tables

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2022-10-04 02_57_49-DBeaver Ultimate 22 2 1 - measurement_partition

2022-10-04 03_09_12-DBeaver Ultimate 22 2 1 - measurement_partition

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java [hunk 1]
```diff
@@ -85,6 +86,9 @@ protected Composite createDialogArea(Composite parent) {
                 if (supportPermissions && supportComments && supportFullDDL) {
                     break; //it supports everything
                 }
+                if (sourceObject.supportsObjectDefinitionOption(DBPScriptObject.OPTION_INCLUDE_PARTITIONS)) {
+                    supportsPartitionsDDL = true;
+                }
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java:91]: Maybe `supportsPartitionsDDL = sourceObject.supportsObjectDefinitionOption(DBPScriptObject.OPTION_INCLUDE_PARTITIONS)`?

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/internal/SQLEditorMessages.properties [hunk 2]
```diff
@@ -319,6 +320,8 @@ source_viewer_separate_fk_text = Separate FK
 source_viewer_separate_fk_tip = Generate foreign keys as separate statements
 source_viewer_show_comments_text = Show comments
 source_viewer_show_comments_tip = Show column comments in table definition
+source_viewer_show_partitions_ddl_text = Show partitions
+source_viewer_show_partitions_ddl_tip = Show partitions tables DDL
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/internal/SQLEditorMessages.properties]:
```suggestion
source_viewer_show_partitions_ddl_tip = Show partitioned tables DDL
```

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/generator/SQLGeneratorDialog.java:91]: Well, I see that all code in this method is written this way. Okay

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/internal/SQLEditorMessages.properties (hunk 2 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/internal/SQLEditorMessages.properties]: Thanks

**Review by E1izabeth — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**